### ISSUE TITLE
Jellyfin won't let Nvidia Shield Energy Saver (sleep timer) setting work

### ISSUE THREAD

**Issue by TorrentFiend** [jellyfin/jellyfin-androidtv#4007]
### Describe the bug

Jellyfin Client on Nvidia Shield seems to ignore my setting for Energy Saver preferences.  To be clear I have my sleep timer set for 1 hour and this with the CEC features my TV has works great as a sleep timer.  It is a frequent thing that I fall asleep watching things and it is great knowing that my nvidia shields sleep mode will also turn off the soundbar and TV all at once after 1 hour of no activity.  If I can't stay awake for the rest of a movie or show I wake up a few hours later and my TV and everything turns itself off instead of everything running all night wasting electricity and hours of backlight use on the display.  Even worse this is a BIG issue on OLEDs where burn in can happen if static images are on too long.  

I have tested this on Emby, Plex, Nova Video Player, SmartTube, Netflix, Max, hulu, amazon prime.... etc etc.  They all work fine and no other app on my shield keeps my display on forever but if I use Jellyfin to watch anything and fall asleep it stays on until I eventually find it still on the next morning and manually shut it off wasting away for hours needlessly.  It will be completely idle on my Jellyfin page of whatever the last thing watched that ended was.

I have tested this several times simply by opening any page within Jellyfin, for example just opening the app and walking away....it will stay on all day and never goes to sleep like it should.  This has meant for now I am reluctantly using Emby still until this is hopefully fixed.  If I fall asleep using Emby the shields sleep timer functions normally and I can trust it won't stay on all night.

I've also tested this turning on the option under energy saver preferences to "Allow turning screen off during media playback" as a TEST ONLY and this DOES allow it to turn off as it should but I never would want to use it this way since if a file is playing from a spinning drive and the display suddenly stops and turns off mid playback I fear it may be bad on the drive to abruptly stop playback like this suddenly.  This will act as an override on this setting I guess to turn the TV off NO MATTER WHAT but applications shouldn't force users to resort to this setting right?  I don't want it to shut off if media is playing, only during the idle time after something has ended. Like if I fall asleep the last 20 minutes of a movie, the energy saver timer can take it from there and turn it off while jellyfin is open but IDLE after the credits end.  Under normal working functionality this overkill option shouldn't need to be enabled, nothing else to date has needed this on, Jellyfin shouldn't either right?

I asked this a few days ago on jellyfin discord and someone mistakenly thought I was talking about it sleeping while something is playing so I will be clear I am not meaning WHILE something plays.....I mean after that thing stops playing upon ending, the energy saver settings should then kick in 1 hour after it ends while the device (and I) are idle.  As I said Emby, Plex, and Nova Video Player all allow this setting to kick in.

To recreate and test this should be very easy.  Choose a Energy saver timer setting preference under device preferences in the nvidia shields settings, simply open Jellyfin Client on the shield and wait....it will stay on indefinitely instead of turning itself off after the time length enabled in energy saver preferences.

I really want to stop using Emby and this is the only reason at the moment I can't.  I look forward to this getting fixed soon so I can go full Jellyfin.

### Application version

0.17.5

### Where did you install the app from?

Google Play

### Device information

Nvidia Shield 2017

### Android version

Android TV 9.1.1 Android 11 I believe, the latest available nvidia shield OS update

### Jellyfin server version

10.9.11

**Comment by emadellaloggia** [comment]
try this

https://mega.nz/file/1vkGlJTL#yU-SJLlIilF0mx5qwpPOd3BM8iKZqOJLEQUStR76rrQ
I put the necessary dlls in the archive

**Comment by nielsvanvelzen** [comment]
You can disable the in-app screensaver from the app preferences.

**Comment by TorrentFiend** [comment]
What I posted has nothing to do with the screensaver. I know very well how to operate my device and how to control the preferences and settings. This has nothing to do with the screensaver setting. I'm specifically talking about the energy saver setting which acts like a sleep timer because the device will turn itself off at a certain point after being idle for a duration that you set.

Please read what I posted carefully, I clearly described how you can recreate exactly the scenario for your own testing and verification that it's not working properly. Simply go to your ENERGY SAVER preferences, not screensaver preferences, energy saver preferences it's a different thing.  Select a time length you want the device to go to sleep at, and then simply open Jellyfin and you will see it will not allow your energy saver time preference to kick in when it should. The device stays on all night even if I fall asleep watching something and it wastes a lot of electricity while doing so. The app will not allow the energy saver time preference to kick in allowing the device to go to sleep unless you go to the drastic measure of telling it to turn off even if media is playing, which is an additional setting also found under energy saver settings that can be enabled but normally never needs to be when apps work correctly.  Unfortunately at the moment this app is not working correctly so the only way to force it to turn off no matter what is to enable this setting but it shouldn't need to be used if it would work correctly.  I don't want it to turn off while media is playing, I simply want it to function properly so jellyfin can tell when the device has been idle, and allow it to turn off after a set time of being idle, instead it's like jellyfin thinks it's active at all times even if someone has fallen asleep while browsing their media inside the app. The application does not properly detect the difference between being idle with no activity and something being watched. It simply stays on at all times forever simply because the app is open. It is not properly detecting any idle activity whatsoever but it should.

Please respond when you understand the issue I'm actually talking about, it has nothing to do with screensavers.

How does jellyfin currently detect idle activity when people are using the app? Can it detect idle activity at all if the app is open or does it just think it's always being used at all times so it stays on forever? I understand not wanting the device to go into sleep mode if you are watching a long three and a half hour movie for example, but when that movie playback ends if someone has fallen asleep while watching such a long movie which is something that will happen under normal usage, does the device then recognize idle activity if the app is still open? This is the issue at hand, not screensavers.

**Comment by TorrentFiend** [comment]
> You can disable the in-app screensaver from the app preferences.

The problem I'm talking about does not have anything to do with a screensaver setting. It's a different setting called energy saver or basically sleep mode which allows the device to go to sleep if it's idle. The problem is jellyfin will not detect idle activity. This means the setting never kicks in and the device stays awake all night instead of going to sleep after an hour like I want it to.

So the problem is, how does jelly fin detect idle activity from the app being used? If the media that was playing has ended, it is then idle, does jellyfin know the difference? Please make the app able to detect idle activity so that settings on my Nvidia shield will work correctly.

If you are confused about what I've mentioned here, please get someone on the Android TV development team who understands what I've mentioned. I'm sure someone must.

**Comment by nielsvanvelzen** [comment]
I'm not interested in discussing this. I've told you how to prevent this behavior and you then tell me, the maintainer of the app, that I'm wrong, don't understand your question and should get someone from the team (me)?

Closing as wontfix.

**Comment by TorrentFiend** [comment]
You would think the maintainer of the app would clearly understand the difference between screensaver and energy saver and understand they are two different things and I was not talking about screensaver at all.

I tried what you are suggesting, under screensaver options I told it to turn the device off after 5 minutes, I disabled the energy saver option as well so after 5 minutes screensaver option should turn it off, yet it did not. This means it is a legitimate issue and even with your twisted use this option that isn't the problem to begin with your solution did not work even though you are the maintainer of the app so apparently know everything. I mean this is like taking your car into the shop needing an oil change and they tell you that the problem is you need more windshield washer fluid. You're like wait what about the oil that needs changed, and they're like no do this instead.

Frankly I'm shocked by the dismissive and angry attitude about this legitimate issue with the app because it interferes with normal device functionality and does not allow the energy saving timer to activate while the app is open. Which allows it to run at all times never letting your device turn off. Maybe this is how you use devices but when I'm not using them or maybe have fallen asleep while watching something I want the device to turn off for the rest of the night.

Since you've made it very clear this will not be addressed however then I guess I'll stick with Emby, their UI is better anyway.

Perhaps more than one person should help with issues like this in the future so a strong dictator like opinion is not weaponized at users trying to help make the app better. Maintainer of the app or not you lack the ability to understand legitimate issues people are trying to help with, climb down off your high horse and reread what I said looking at the options I mentioned and you will see I was correct. Again here's a tip, it has nothing to do with the screensaver option.

Apologies for using too many words but with this sort of response, I was trying to explain again what wasn't read the first time.

**Comment by nielsvanvelzen** [comment]
I've already told you how to fix this and you refuse to acknowledge my answer. That's up to you. Call me a dictator all you want but you're only embarrassing yourself.

**Comment by TorrentFiend** [comment]
Again apparently you don't read what I post. I told you I tried your suggested fix and it did not fix it. You refuse to acknowledge that this has nothing to do with the screensaver setting. This is not related to the screensaver setting at all, only you said that. I'm telling you this is related to the energy saver option. I don't know if you have an Nvidia shield but if you do please set your energy saver option and notice it works with any other app, then open jellyfin and watch it not work.

Only then will you understand what I'm talking about.

I've posted this multiple times and I'm tired of repeating myself. If you will not listen when I'm trying to help make this app better because that is my goal, I like this app and would like to help get this issue fixed. I won't post again but no matter how many times you repeat that you told me such and such, you're telling me a fix for a problem I'm not having, you're not listening to what the actual problem is. Please calm down take a step back and try what I suggest and you will see there is indeed a problem with something completely unrelated to screensaver options.

**Comment by maru801** [comment]
> What I posted has nothing to do with the screensaver. I know very well how to operate my device and how to control the preferences and settings. This has nothing to do with the screensaver setting. I'm specifically talking about the energy saver setting which acts like a sleep timer because the device will turn itself off at a certain point after being idle for a duration that you set.
> 
> Please read what I posted carefully, I clearly described how you can recreate exactly the scenario for your own testing and verification that it's not working properly. Simply go to your ENERGY SAVER preferences, not screensaver preferences, energy saver preferences it's a different thing. Select a time length you want the device to go to sleep at, and then simply open Jellyfin and you will see it will not allow your energy saver time preference to kick in when it should. The device stays on all night even if I fall asleep watching something and it wastes a lot of electricity while doing so. The app will not allow the energy saver time preference to kick in allowing the device to go to sleep unless you go to the drastic measure of telling it to turn off even if media is playing, which is an additional setting also found under energy saver settings that can be enabled but normally never needs to be when apps work correctly. Unfortunately at the moment this app is not working correctly so the only way to force it to turn off no matter what is to enable this setting but it shouldn't need to be used if it would work correctly. I don't want it to turn off while media is playing, I simply want it to function properly so jellyfin can tell when the device has been idle, and allow it to turn off after a set time of being idle, instead it's like jellyfin thinks it's active at all times even if someone has fallen asleep while browsing their media inside the app. The application does not properly detect the difference between being idle with no activity and something being watched. It simply stays on at all times forever simply because the app is open. It is not properly detecting any idle activity whatsoever but it should.
> 
> Please respond when you understand the issue I'm actually talking about, it has nothing to do with screensavers.
> 
> How does jellyfin currently detect idle activity when people are using the app? Can it detect idle activity at all if the app is open or does it just think it's always being used at all times so it stays on forever? I understand not wanting the device to go into sleep mode if you are watching a long three and a half hour movie for example, but when that movie playback ends if someone has fallen asleep while watching such a long movie which is something that will happen under normal usage, does the device then recognize idle activity if the app is still open? This is the issue at hand, not screensavers.

You could have responded with "can you explain why you're saying I can disabled the screensaver?" instead of instantly writing this huge block of text going off on how your issue is not being understood.

There's a perfectly good explanation on why that was recommended for you, and if you don't believe it's relevant, all you have to do is nicely ask for clarification on why that is. The lead maintainer of this app was put off on responding to you due to how this escalated.

**Comment by TorrentFiend** [comment]
I don't want to upset anyone, I have good intentions of trying to make this app better with an issue that I noticed under normal use. I clearly know how to recreate this and I tried to communicate that. Please everyone calm down if that's possible.

If I over posted it's because I felt like I was repeating myself or being misunderstood. There is an issue though and I wish it would be acknowledged and hopefully someday fixed.

Everyone have a good day.

**Comment by nielsvanvelzen** [comment]
# GUIDE: Disabling the in-app screensaver in Jellyfin for Android TV

[91700003fe.webm](https://github.com/user-attachments/assets/05f9467f-3cf9-4319-8191-734ddb2401c6)

_note: the screenrecording does not include the process of adding your server and signing in. These steps must be taken first before proceeding with this guide_

**Comment by TorrentFiend** [comment]
> # 🎥 GUIDE: Disabling the in-app screensaver in Jellyfin for Android TV
>  91700003fe.webm 
> Important
> 
> The screen recording does not include the process of adding your server and signing in. These steps must be taken first before proceeding with this guide

Okay my bad, I have to apologize on this one. Once you posted that video I realized there was a setting in the app preferences I didn't know about related to the screensaver. When you mentioned screensaver options I was certain you were talking about the device preferences of The shield hardware and it's screensaver options, I didn't know you meant the in-app settings but you did clearly State an app preferences on your response, sorry about that.

I just tried setting the jelly thin screensaver for 5 minutes and the sleep timer energy saver option for 20 minutes with that preference unchecked on the app and it did in fact go off like it should. It appears to work. Thank you for clarifying what you meant through video to help me out on that, I really appreciate it and can continue happily enjoying this wonderful app.

I'm a little confused why there is a separate option inside the jellyfin regarding the screensaver that can cause such an unwanted scenario to happen at all because I'm sure a lot of users probably miss that small detail inside the app that will break this functionality but again thank you for spelling this one out.

It's working correctly now. I agree this matter is closed, thanks again, even if we butted heads getting there that was mostly my fault so I apologize.
